cardano/src/block/block.rs

```diff
@@ -295,6 +368,19 @@ impl cbor_event::de::Deserialize for Block {
     }
 }
 
+impl<'a> cbor_event::se::Serialize for BlockHeaderView<'a> {
+    fn serialize<'se, W: Write>(
+        &self,
+        serializer: &'se mut Serializer<W>,
+    ) -> cbor_event::Result<&'se mut Serializer<W>> {
+        let serializer = serializer.write_array(cbor_event::Len::Len(2))?;
+        match self {
+            BlockHeaderView::Boundary(hdr) => serializer.write_unsigned_integer(0)?.serialize(hdr),
+            BlockHeaderView::Normal(hdr) => serializer.write_unsigned_integer(1)?.serialize(hdr),
+        }
+    }
+}
+
```

Comment:
can you add a test for the BlockHeaderView serialize actually deserialize to the expected BlockHeader ?

Reply:
Good idea, will do. Should I use property test grinding?